### PR TITLE
dev to main sync

### PR DESCRIPTION
**Pull request by shubhamsigdar1**
**Date: 30-05-2023**
**Developer Name:** @shubhamsigdar1 , @sahsisunny , @khushi818 , @RitikJaiswal75, @kotesh-arya 

**PR Number(s):-** 
- #580
- #571
- #522
- #599

**Issue Ticket Number:-** 
- #572
- #587
- #579
- #503
- #466

**Backend changes**

- [x] Yes [Backend PR](https://github.com/Real-Dev-Squad/website-backend/pull/1099) [Frontend PR](#599)
- [ ] No

**Frontend Changes**

- [x] Yes
- [ ] No

**Database changes**

- [ ] Yes
- [x] No

**Breaking changes (If your feature is breaking/missing something please mention pending tickets)**

- [x] No
- [ ] Yes:- 

**Deployment notes(Any special instructions for the deployment)**

N/A

**Tested in staging**

- [x] Yes
- [ ] No

**List of PRs going in this Sync.** 
- [x] https://github.com/Real-Dev-Squad/website-status/pull/595
- [x] #580 
- [x] #571 
- [x] https://github.com/Real-Dev-Squad/website-status/pull/522
- [x] #599 

## Screenshots.
**Feature: Markdown to HTML**

image
image

**Fixed task card long title(after adding feature-flag)**

image

**Feat: width-tabs**

width
image
image

**REFACTOR: API calls for TaskDetails Page Migrated to RTK Query **

https://github.com/Real-Dev-Squad/website-status/assets/69259490/308d4d42-5767-4c0c-8207-2af5cdeaae3a

### CODE REVIEW

**Comment by RitikJaiswal75** [comment]
please add the screenshot of the width fix with the URL bar showing, else how will I know if it is on staging or local

**Comment by khushi818** [comment]
> please add the screenshot of the width fix with the URL bar showing, else how will I know if it is on staging or local

yup I have added

**Comment by vinit717** [comment]
For task Dependency 
image
image

**Comment by RitikJaiswal75** [comment]
** REFACTOR: API calls for TaskDetails Page Migrated to RTK Query **
Please use the video for above in staging not on dev

File: src/app/services/taskDetailsApi.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { TASKS_URL } from '@/constants/url';
+import { api } from './api';
+import { taskDetailsDataType } from '@/interfaces/taskDetails.type';
+
+export const taskDetailsApi = api.injectEndpoints({
```
Comment (iamitprakash) [inline src/app/services/taskDetailsApi.ts:5]: updateTaskDetails mutation is incorrectly using the PATCH method, when it should be using the PUT method. The PATCH method is used to update a specific resource, while the PUT method is used to replace a resource entirely. In this case, we want to replace the entire task details resource, so we should be using the PUT method.

**Review by iamitprakash — CHANGES_REQUESTED** [review]

**Pull request merged**